cryptodoc/src/08_signatures.rst

```diff
@@ -334,15 +334,17 @@ The signature generation algorithm works as follows:
    3. Compute point :math:`W=(x_1,y_1)=k'*G`. This computation utilizes randomized Jacobian point
       coordinates with a blinding masks that is equal in size to the
       underlying field.
-   4. Compute
-      :math:`{r = H}{(x_{1})}`
-      , where :math:`H`
+   4. Compute the witness
+      :math:`{r = h}{(x_{1})}`
+      , where :math:`h`
       is the hash function used in the current instance of the EMSA1
       signature scheme.
-   5. Compute
+   5. If the output length of the hash function :math:`h` exceeds the size of the group order,
```

Comment:
I looked at ISO 14888-3 and somehow I can't finde where this is specified in the standard. I was under the impression that I saw it before... can you point me at the relevant line?

Reply:
OK, but then I think this should be made a bit more precise here. 
I guess it should really give a formula like in the standard. The way it is formulated here is confusing. (Truncate the *most significant bytes* could even mean that these are omitted and the least significant bytes are left over.) Without a formula you also probably don't know if you need to use ceil or floor ;-)

Reply:
We can certainly improve the wording if it is unclear/confusing.

But in general I think it would make sense to try to keep all this algorithm description not to formal and detailed, and see them more as a bridge between the specification and the implementation.
With the important details and differences to the specification highlighted.

If one needs to know each single detail how it is implement in Botan, one should look at the code. Everything written here will be more imprecise.

If one wants to know how an algorithm should be implemented, they should look at the original specification. Repeating the complete specification here in our own words does not make sense to me.

Reply:
Changed it a little, hope it is more clear now.